lib/CMakeLists.txt

```
@@ -122,18 +122,22 @@ if(NOT DEFINED SHARE_LIB_OBJECT)
   endif()
 endif()
 
+if(WIN32)
+  # On Windows we handle exports via libcurl.def. Except when symbol hiding is
+  # disabled or debug mode is enabled. In those cases we export all symbols
+  # from libcurl DLL, and skip using the .def file.
```

i suggest really spell it out, for example

```suggestion
  # from libcurl DLL, and skip using the .def file.
  # Define CURL_STATICLIB always, to disable __declspec(dllexport) for exported
  # libcurl symbols.
```

otherwise i think for most people reading it it's not immediately obvious how CURL_STATICLIB relates

---

Agreed. How does this version look?:
```
  # Define CURL_STATICLIB always, to disable __declspec(dllexport) for exported
  # libcurl symbols. We handle exports via libcurl.def instead. Except when
  # symbol hiding is disabled or debug mode is enabled. In those cases we
  # export _all_ symbols from libcurl DLL, without using the .def file.
```

---

looks good